docs/module-policy.rst

```diff
@@ -0,0 +1,49 @@
+Module inclusion policy
+=======================
+
+Python is a dynamic language with a complex module system, including
+modules that are created only at runtime or appear on specific
+supported platforms.
+
+This page exists to document ``stdlib-list``'s approach to module detection
+and subsequent inclusion. It is not intended to be permanent, and may change
+over time as Python itself changes (or our approach to module detection
+improves).
+
+Current guiding rules
+---------------------
+
+* Missing top-level modules **are a bug**: if a new version of Python adds a new
+  top-level module, our failure to detect it should be considered a bug.
+
+  Concretely: if ``examplemodule`` is present in Python 3.999, then it should be
+  included in the ``stdlib_list("3.999")`` listing.
+
+* Missing sub-modules are **best-effort**: if ``examplemodule`` contains
+  ``examplemodule.foo.bar.baz.deeply.nested``, we make a best-effort attempt
+  to detect each inner module but make no guarantee about doing so.
+
+  Our rationale for this is that "stdlib-ness" is inherited from the parent
+  module, even when not explicitly listed. In other words: anything that matches
+  ``examplemodule.*`` is in the standard library by definition so long
+  as ``examplemodule`` is in the standard library.
```

Comment:
This rule has made me realize that we should maybe tweak the behavior of `in_stdlib` -- `foo.bar.baz` is in the stdlib by definition if `foo` is, so instead of checking whether an exact submodule exists we should maybe just return `True` if the parent module(s) are ones we know.

OTOH, this has weird consequences for our semantics: `in_stdlib` then essentially becomes a predicate for "anything that might be under a stdlib module namespace," not "this specific submodule path definitely exists." Which is undesirable.

Reply:
Would this change the behavior for folks adding their own pseudo modules under a stdlib module? (Let's ignore if that's a good idea or not.)

For example, if I add my own `io.mike` module, I would not expect a check for it to ever return True from `in_stdlib`

Reply:
It would, yeah, which is one of the concerns I have with it.

Rather than changing `in_stdlib`, maybe we should have a new complementary API like `in_stdlib_namespace` -- that would essentially be a superset of `in_stdlib`, except that it won't require that the identifier actually exists/is a module. I think that would also handle the pseudo-module problem well. WDYT?

Concretely:

```
>>> in_stdlib("io.mike")
False
>>> in_stdlib_namespace("io.mike")
True
```

Reply:
Sounds good! I'll file a separate tracking issue for that new API.

Reply:
#120